src/Uno.UI/UI/Xaml/Controls/Repeater/ElementManager.cs

```diff
@@ -0,0 +1,506 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using static Microsoft.UI.Xaml.Controls._Tracing;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	internal static class CollectionExtensions
+	{
+		public static bool TryGetElementAt<T>(this IList<T> collection, int index, out T element)
+			where T : class
+		{
+			if (index < collection.Count)
+			{
+				element = collection[index];
+				return element != default; // TODO UNO
+			}
+			else
+			{
+				element = default;
+				return false;
+			}
+		}
+
+		public static void AddOrInsert<T>(this IList<T> collection, int index, T element)
+		{
+			if (index < collection.Count)
+			{
+				collection.Add(element);
+			}
+			else
+			{
+				collection.Insert(index, element);
+			}
+		}
+	}
+
+	internal class ElementManager
+	{
+		private bool IsVirtualizingContext
+		{
+			get
+			{
+				if (m_context != null)
+				{
+					var rect = m_context.RealizationRect;
+					bool hasInfiniteSize = double.IsInfinity(rect.Height) || double.IsInfinity(rect.Width);
+					return !hasInfiniteSize;
+				}
+
+				return false;
+			}
+		}
+
+		public int GetRealizedElementCount => IsVirtualizingContext ? (int)(m_realizedElements.Count) : m_context.ItemCount;
+
+		private readonly List<UIElement> m_realizedElements = new List<UIElement>();
+		private readonly List<Rect> m_realizedElementLayoutBounds = new List<Rect>();
+		private int m_firstRealizedDataIndex = -1;
+		private VirtualizingLayoutContext m_context;
+
+		public void SetContext(VirtualizingLayoutContext virtualContext)
+		{
+			m_context = virtualContext;
+		}
+
+		public void OnBeginMeasure(ScrollOrientation orientation)
+		{
+			if (m_context != null)
+			{
+				if (IsVirtualizingContext)
+				{
+					// We proactively clear elements laid out outside of the realizaton
+					// rect so that they are available for reuse during the current
+					// measure pass.
+					// This is useful during fast panning scenarios in which the realization
+					// window is constantly changing and we want to reuse elements from
+					// the end that's opposite to the panning direction.
+					DiscardElementsOutsideWindow(m_context.RealizationRect, orientation);
+				}
+				else
+				{
+					// TODO UNO
+					//// If we are initialized with a non-virtualizing context, make sure that
+					//// we have enough space to hold the bounds for all the elements.
+					//int count = m_context.ItemCount;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5863680)